hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -2054,10 +2054,10 @@ public void acquireLock(TransactionLocks locks) throws IOException {
             .setNameNodeID(nameNode.getId())
             .setActiveNameNodes(nameNode.getActiveNameNodes().getActiveNodes())
             .skipReadingQuotaAttr(!dir.isQuotaEnabled());
-        locks.add(il)
-            .add(lf.getBlockLock())
-            .add(lf.getLeaseLock(LockType.WRITE, holder))
-            .add(lf.getLeasePathLock(LockType.READ_COMMITTED)).add(
+        locks.add(il).add(lf.getBlockLock());
+        LeaseLock leaseLock = (LeaseLock)lf.getLeaseLock(LockType.WRITE, holder);
```

Comment:
Related to my other comment, i think it is better to add another getLeaseLock that takes locktype, holder, and src as arguments and remove the setter for singleFileLock

Reply:
I agree.

Reply:
I agree, more work for me